homeassistant/components/http/ban.py

```diff
@@ -241,8 +275,29 @@ def _add_ban(self, ip_ban: IpBan) -> None:
             # Write in a single write call to avoid interleaved writes
             out.write("\n" + yaml.dump(ip_))
 
+    def _write_bans(self) -> None:
+        """Write IP bans to config file."""
+        ip_bans = {
+            str(ip_ban.ip_address): {ATTR_BANNED_AT: ip_ban.banned_at.isoformat()}
+            for ip_ban in self.ip_bans_lookup.values()
+        }
+
+        with open(self.path, "r+", encoding="utf8") as out:
+            out.seek(0)
+            out.truncate()
+            out.write("\n" + yaml.dump(ip_bans))
+
     async def async_add_ban(self, remote_addr: IPv4Address | IPv6Address) -> None:
         """Add a new IP address to the banned list."""
         if remote_addr not in self.ip_bans_lookup:
             new_ban = self.ip_bans_lookup[remote_addr] = IpBan(remote_addr)
             await self.hass.async_add_executor_job(self._add_ban, new_ban)
+
+    async def async_remove_ban(self, remote_addr: IPv4Address | IPv6Address) -> None:
+        """Remove an IP address from the banned list."""
+        if remote_addr in self.ip_bans_lookup:
+            self.ip_bans_lookup.pop(remote_addr)
+            await self.hass.async_add_executor_job(self._write_bans)
```

Comment:
When `_write_bans` runs in a thread it will observe the latest state of `self.ip_bans_lookup`, which by that point in time may be different to what it was when line 298 runs.

I don't think there are any serious problems (race conditions) that can come of it but thought I would mention it since it's different to `async_add_ban`, which passes the IP to add by value.